internal/grpc/services/gateway/usershareprovider.go

```diff
@@ -76,20 +85,32 @@ func (s *svc) CreateShare(ctx context.Context, req *collaboration.CreateShareReq
 			if denyGrantStatus.Code != rpc.Code_CODE_OK {
 				return &collaboration.CreateShareResponse{
 					Status: denyGrantStatus,
-				}, err
+				}, nil
+			}
+		} else {
+			addGrantStatus, err := s.addGrant(ctx, req.ResourceInfo.Id, req.Grant.Grantee, req.Grant.Permissions.Permissions)
+			if err != nil {
+				log.Error().Err(err).Str("Message", addGrantStatus.Message).Msg("Failed to Create Share: error during addGrant")
```

Comment:
Good for the log, let's add also `req.ResourceInfo` and `req.Grant.Grantee` to the message